ludwig/features/image_feature.py

```diff
@@ -241,16 +243,56 @@ def _renormalize_image(self, images):
             return images.type(torch.float32).div(255.0)
 
 
+def _get_torchvision_transform(torchvision_parameters: TVModelVariant) -> torch.nn.Module:
+    """Returns a torchvision transform that is compatible with the model variant.
+
+    Note that the raw torchvision transform is not returned. Instead, a Sequential module that includes
+    image resizing is returned. This is because the raw torchvision transform assumes that the input image has
+    three channels, which is not always the case with images input into Ludwig.
+
+    Args:
+        torchvision_parameters: The parameters for the torchvision model variant.
+    Returns:
+        (torchvision_transform, transform_metadata): A torchvision transform and the metadata for the transform.
+    """
+    torchvision_transform_raw = torchvision_parameters.model_weights.DEFAULT.transforms()
+    torchvision_transform = torch.nn.Sequential(
+        ResizeChannels(num_channels=3),
+        torchvision_transform_raw,
+    )
+    transform_metadata = {
+        "height": torchvision_transform_raw.crop_size[0],
+        "width": torchvision_transform_raw.crop_size[0],
+        "num_channels": len(torchvision_transform_raw.mean),
+    }
+    return (torchvision_transform, transform_metadata)
+
+
+def _get_torchvision_parameters(model_type: str, model_variant: str) -> TVModelVariant:
+    return torchvision_model_registry.get(model_type).get(model_variant)
+
+
 class _ImagePreprocessing(torch.nn.Module):
     """Torchscript-enabled version of preprocessing done by ImageFeatureMixin.add_feature_data."""
 
-    def __init__(self, metadata: TrainingSetMetadataDict, tv_transforms: Optional[torch.nn.Module] = None):
+    def __init__(
+        self,
+        metadata: TrainingSetMetadataDict,
+        torchvision_transform: Optional[torch.nn.Module] = None,
+        transform_metadata: Optional[Dict[str, Any]] = None,
```

Comment:
nit: Consider breaking up `transform_metadata` into separate fields: `transform_height`, `transform_width, `transform_num_channels`, or used a dataclass for stronger field guarantees and typing:

```python
@dataclass
class ImageTransformMetadata:
    height: int
    width: int
    num_channels: int
```

Reply:
nice suggestion– I wasn't sure that they supported this, but apparently they do! https://github.com/pytorch/pytorch/issues/72901